include/mbgl/util/thread.hpp

```diff
@@ -40,24 +40,33 @@ template<class Object>
 class Thread : public Scheduler {
 public:
     template <class... Args>
-    Thread(const std::string& name, Args&&... args) {
-        std::promise<void> running;
-
-        thread = std::thread([&] {
+    Thread(const std::string& name, Args&&... args)
+        : object(std::make_unique<Actor<Object>>()) {
+
+        std::unique_ptr<std::promise<void>> running_ = std::make_unique<std::promise<void>>();
```

Comment:
This doesn't need to be a `unique_ptr`. Instead, create `std::promise<void> running_` on the stack, `std::move` it into the lambda captures, and give the lambda the [`mutable`](https://en.cppreference.com/w/cpp/language/lambda) specifier so that it can call the non-const `set_value` method.